### ISSUE TITLE
Multiple providers error when defining custom params for consume with BOSH 266

### ISSUE THREAD

**Issue by bandesz** [cloudfoundry/bosh#1990]
## Environment

BOSH 266.4.0
Cloud Foundry: cf-deployment v1.38.0
Using BOSH DNS: no
Using Consul DNS: yes

## What is the issue

We started to use log-cache (https://github.com/cloudfoundry/log-cache-release) which depends on BOSH links and consumes two jobs: reverse_log_proxy and cloud_controller.

As we don't use BOSH DNS yet we had to override the addresses and the ports for these services.

```
- type: replace
  path: /instance_groups/name=doppler/jobs/name=log-cache-nozzle/consumes/reverse_log_proxy
  value:
    address: reverse-log-proxy.service.cf.internal
    instances: []
    properties:
      reverse_log_proxy:
        egress:
          port: 8082

- type: replace
  path: /instance_groups/name=doppler/jobs/name=log-cache-cf-auth-proxy/consumes?/cloud_controller?
  value:
    address: cloud-controller-ng.service.cf.internal
    instances: []
```

This all worked fine while using BOSH 265.2.0, but started to throw the following error with 266.4.0:

```
- Multiple providers of type 'reverse_log_proxy' found for consumer link 'reverse_log_proxy' in job 'adapter' in instance group 'adapter'. All of these match:
   Deployment: foo, instance group: log-api, job: reverse_log_proxy, link name/alias: reverse_log_proxy
   Deployment: foo, instance group: doppler, job: log-cache-nozzle, link name/alias: reverse_log_proxy
  - Multiple providers of type 'cloud_controller' found for consumer link 'cloud_controller' in job 'scheduler' in instance group 'scheduler'. All of these match:
   Deployment: foo, instance group: doppler, job: log-cache-cf-auth-proxy, link name/alias: cloud_controller
   Deployment: foo, instance group: api, job: cloud_controller_ng, link name/alias: cloud_controller
```

Is this a bug or are we doing the address override wrong?

Even if we don't use BOSH DNS the addresses for the links still follow a BOSH DNS pattern, e.g. `q-s0.api.cf.foo.bosh` - how can we configure BOSH to emit the Consul domain names?

**Comment by pivotal-jamil-shamy** [comment]
@bandesz thanks for reporting this issue. Looks like the manual links introduced under `name=log-cache-nozzle/consumes/reverse_log_proxy` and `name=log-cache-cf-auth-proxy/consumes?/cloud_controller?` are also registering themselves as providers that cause this issue.

We will investigate more and submit a fix (if needed). [Tracker Story](https://www.pivotaltracker.com/story/show/158630440) for the issue.

**Comment by pivotal-jamil-shamy** [comment]
@bandesz The issue should is fixed in [266.6.0 release](https://github.com/cloudfoundry/bosh/releases/tag/v266.6.0).
Let us know if that works for you.
